types/address/hash.go

```diff
@@ -91,3 +95,121 @@ func Module(moduleName string, derivationKeys ...[]byte) []byte {
 func Derive(address, key []byte) []byte {
 	return Hash(conv.UnsafeBytesToStr(address), key)
 }
+
+// HeimdallHash represents heimdall address
+type HeimdallHash common.Hash
```

Comment:
Seems like this type and its methods are to be used instead some of the (pre-existing) functions defined above. Is that correct?
If yes, it would be good to build and unit-test while temporarily removing the functions above no longer needed: if you get build errors, they point to places we still need to adapt to Heimdall hashes.

Reply:
Not sure if those methods will eventually replace the ones defined above, to be honest.
`HeimdallHash` was imported as part of this file in the current `heimdall` implementation, but it has no usage so far.
Will double check this, and eventually implement unit test to remove the pleonastic methods.

Reply:
Not addressed yet

Reply:
So I checked the code and apparently `HeimdallHash` is used in `staking` module, being worked on by @VAIBHAVJINDAL3012 in `heimdall-v2` repo.
I'd leave this as is, until Vaibhav is done, then we can double check usage of these methods. Btw, @VAIBHAVJINDAL3012 is using this branch from cosmos, so he has these types available.

Reply:
Resolved